File: nfc/src/main/java/com/piotrekwitkowski/nfc/HCE.java
package com.piotrekwitkowski.nfc;

import android.content.Context;

import com.piotrekwitkowski.log.Log;

import java.io.IOException;

import static com.piotrekwitkowski.nfc.Iso7816.wrapApdu;

public class HCE {
    private static final String TAG = "HCE";

    public static Response selectAndroidApp(Context context, IsoDep isoDep) throws IOException {
        Log.i(TAG, "selectAndroidApp()");

        String HCE_AID = context.getString(R.string.hce_aid);
        byte[] commandApdu = wrapApdu(ByteUtils.toByteArray(HCE_AID));
        return isoDep.transceive(commandApdu);
    }
}

File: LibraryReader/src/main/java/com/piotrekwitkowski/libraryreader/HCE.kt
package com.piotrekwitkowski.libraryreader

import android.content.Context
import com.piotrekwitkowski.log.Log.i
import com.piotrekwitkowski.nfc.ByteUtils.toByteArray
import com.piotrekwitkowski.nfc.Iso7816.wrapApdu
import java.io.IOException

internal object HCE {
    private const val TAG = "HCE"

    @Throws(IOException::class)
    fun selectAndroidApp(context: Context, isoDep: IsoDep): Response? {
        i(TAG, "selectAndroidApp()")

        val HCE_AID = context.getString(com.piotrekwitkowski.nfc.R.string.hce_aid)
        val commandApdu = wrapApdu(toByteArray(HCE_AID))
        return isoDep.transceive(commandApdu)
    }
}
